engines/griffon/cutscenes.cpp

```diff
@@ -0,0 +1,553 @@
+
+	do {
+		if (_ticks < ticks1 + 1500) {
+			y1 = 255 * (_ticks - ticks1) / 1500;
+			y1 = CLIP(y1, 0, 255);
+		} else
+			break;
+
+		_videoBuffer->fillRect(Common::Rect(0, 0, _videoBuffer->w, _videoBuffer->h), 0);
+
+		_videoBuffer->setAlpha(y1);
+		_videoBuffer2->blit(*_videoBuffer3);
+		_videoBuffer->blit(*_videoBuffer3);
+
+		g_system->copyRectToScreen(_videoBuffer3->getPixels(), _videoBuffer3->pitch, 0, 0, _videoBuffer3->w, _videoBuffer3->h);
+		g_system->updateScreen();
+
+		if (g_system->getEventManager()->pollEvent(_event)) {
+			if (_event.type == Common::EVENT_QUIT) {
+				_shouldQuit = true;
+				return;
+			}
+		}
+
+		g_system->delayMillis(10);
+
+		_ticksPassed = _ticks;
+		_ticks = g_system->getMillis();
+
+		_ticksPassed = _ticks - _ticksPassed;
+		_fpsr = (float)_ticksPassed / 24;
+
+		_fp++;
+		if (_ticks > _nextTicks) {
+			_nextTicks = _ticks + 1000;
+			_fps = _fp;
+			_fp = 0;
+		}
+	} while (1);
+
+
+	int keywait = 2000 + _ticks;
+
+	ticks1 = _ticks;
+	y1 = 0;
+	do {
+
+		_videoBuffer->copyRectToSurface(_theEndImg->getPixels(), _theEndImg->pitch, 0, 0, _theEndImg->w, _theEndImg->h);
+
+		y1 = 255;
+		if (_ticks < ticks1 + 1000) {
+			y1 = 255 * (_ticks - ticks1) / 1000;
+			y1 = CLIP(y1, 0, 255);
+		}
+
+		_videoBuffer->setAlpha(y1);
+
+		_videoBuffer2->fillRect(Common::Rect(0, 0, _videoBuffer2->w, _videoBuffer2->h), 0);
+		_videoBuffer->blit(*_videoBuffer2);
+
+		g_system->copyRectToScreen(_videoBuffer2->getPixels(), _videoBuffer2->pitch, 0, 0, _videoBuffer2->w, _videoBuffer2->h);
+
+		g_system->updateScreen();
+		g_system->delayMillis(10);
+
+		_ticksPassed = _ticks;
+		_ticks = g_system->getMillis();
+
+		_ticksPassed = _ticks - _ticksPassed;
+		_fpsr = (float)_ticksPassed / 24;
+
+		_fp++;
+		if (_ticks > _nextTicks) {
+			_nextTicks = _ticks + 1000;
+			_fps = _fp;
+			_fp = 0;
+		}
+
+		if (g_system->getEventManager()->pollEvent(_event)) {
+			if (_event.type == Common::EVENT_QUIT) {
+				_shouldQuit = true;
+				return;
+			}
+
+			if (_event.type == Common::EVENT_KEYDOWN && keywait < _ticks)
+				break;
+		}
+
+	} while (1);
+
+	_videoBuffer2->fillRect(Common::Rect(0, 0, _videoBuffer2->w, _videoBuffer2->h), 0);
+	_videoBuffer3->fillRect(Common::Rect(0, 0, _videoBuffer3->w, _videoBuffer3->h), 0);
+
+	theEnd();
+}
+
+void GriffonEngine::theEnd() {
+	for (int i = 0; i < kMaxFloat; i++) {
+		_floatText[i].framesLeft = 0;
+		_floatIcon[i].framesLeft = 0;
+	}
+
+	for (float y = 0; y < 100; y += _fpsr) {
+		_videoBuffer2->fillRect(Common::Rect(0, 0, _videoBuffer2->w, _videoBuffer2->h), 0);
+		_videoBuffer->setAlpha((int)((100.0 - y) / 100 * 255));
+		_videoBuffer->blit(*_videoBuffer2);
+
+		g_system->copyRectToScreen(_videoBuffer2->getPixels(), _videoBuffer2->pitch, 0, 0, _videoBuffer2->w, _videoBuffer2->h);
+		g_system->updateScreen();
+
+		if (g_system->getEventManager()->pollEvent(_event)) {
+			if (_event.type == Common::EVENT_QUIT) {
```

Comment:
You might want to consider doing the quit event check as a macro. It just seems a bit redundant to keep seeing it repeated so many times in this file.

Reply:
Done